Alpaca.Markets/RateThrottler.cs

```diff
@@ -79,16 +99,26 @@ internal sealed class RateThrottler : IThrottler, IDisposable
         /// <inheritdoc />
         public void Dispose()
         {
-            _semaphore.Dispose();
+            _throttleSemaphore.Dispose();
             _exitTimer.Dispose();
         }
 
         /// <inheritdoc />
         public void WaitToProceed()
         {
 
+            // Block until any all stops are over
+            while (_allStop && unchecked(_allStopUntil - Environment.TickCount) > 0)
```

Comment:
Logic issue - Environment.TickCount can be negative and if _allStopUntil will be 0 your implementation will be wrong. I suggest to use DateTime variable instead of this error-prone tick count based approach.

Reply:
_allStop is built the same way the throttle wait logic is setup; an increment on Environment.TickCount.  I tested that use of "unchecked" a fair bit myself when making these changes because I found it hard to believe that it worked, but in all my tests it did, in fact, work out mathematically.  If you've found a scenario where it doesn't work, then it wouldn't work for the throttler either.